[flang] Fix MASKR/MASKL lowering for INTEGER(16)

jeanPerier opened the pull request:
The all one masks was not properly created for i128 types because builder.createIntegerConstant ended-up truncating -1 to something positive.

Add a builder.createAllOnesInteger/createMinusOneInteger helpers and use them where createIntegerConstant(..., -1) was used.
Add an assert in createIntegerConstant to catch negative numbers for i128 type.

jeanPerier committed:
[flang] Fix MASKR/MASKL lowering for INTEGER(16)

The all one masks was not properly created for i128 types because
builder.createIntegerConstant ended-up truncating -1 to something
positive.

Add a builder.createAllOnesInteger helper.

vdonaldson reviewed (approved):
Have you tried using bbc with this change for tests that exercise kind=10 and kind=16? It's not mandatory, but it's nice to keep bbc as usuable as possible. (The lit tests may or may not already do that depending on the types involved.)

jeanPerier committed:
Fix ieee_class regression - use APInt to build masks

The ieee_class was actually relying on createIntegerConstant
to zero extend "-1" from 64 to 128 bits.

Use APInt instead to build the masks, this allows simplifying
the code.

use APInt

jeanPerier commented:
> Have you tried using bbc with this change for tests that exercise kind=10 and kind=16? It's not mandatory, but it's nice to keep bbc as usuable as possible. (The lit tests may or may not already do that depending on the types involved.)

The lit tests are using both bbc -emit-fir/flang-new -fc1 -emit-fir

vdonaldson commented:
> > Have you tried using bbc with this change for tests that exercise kind=10 and kind=16? It's not mandatory, but it's nice to keep bbc as usuable as possible. (The lit tests may or may not already do that depending on the types involved.)
> 
> The lit tests are using both bbc -emit-fir/flang-new -fc1 -emit-fir

Thanks Jean!
